### PR TITLE
Alpha Version - Edit feature

### LINKED ISSUE #75: Alpha Version - Edit feature
- Make `edit` work with new fields.
- Make `StudyGroupTags` persistent.

### PR DESCRIPTION
**Pull request by wkxcass**
Fixes #75 

### - `edit` now works with new fields (i.e., `gender`, `age` and `detail`)

1. Successful executions:
Screenshot 2024-10-20 224756
Screenshot 2024-10-20 224907
_p.s. `edit <index> d/` will set `detail` into an **empty String**_

2. Exceptional event:
Screenshot 2024-10-20 224828

### - For `StudyGroupTag`s, input values are now appended to the existing tag set (instead of overwriting).

1. Input: brand new `StudyGroupTag` _(initial tags: "1A", "Control")_
Screenshot 2024-10-20 224940

2. Input: one brand new and one existing `StudyGroupTag` _(initial tags: "1A", "1B", "Control")_
Screenshot 2024-10-20 225022

3. Input: existing `StudyGroupTag`s _(initial tags: "1A", "1B", "Completed", "Control")_
Screenshot 2024-10-20 225111

### - Also refactored and standardized some code along the way~

### CODE REVIEW

**Review by k-hian — COMMENTED** [review]
LGTM for now.

File: src/test/java/seedu/address/logic/commands/EditCommandTest.java [hunk 1]
```diff
@@ -37,14 +37,17 @@ public class EditCommandTest {
...
 
+        Person personToEdit = model.getFilteredPersonList().get(0);
+        model.setPerson(personToEdit, new PersonBuilder(personToEdit).withStudyGroupTags().build());
+
         Model expectedModel = new ModelManager(new AddressBook(model.getAddressBook()), new UserPrefs());
-        expectedModel.setPerson(model.getFilteredPersonList().get(0), editedPerson);
+        expectedModel.setPerson(expectedModel.getFilteredPersonList().get(0), editedPerson);
 
         assertCommandSuccess(editCommand, model, expectedMessage, expectedModel);
     }
```
Comment (wkxcass) [inline src/test/java/seedu/address/logic/commands/EditCommandTest.java:53]: Managed to work around this testcase as above for now, open to any improvement suggestions!

Comment (wkxcass) [inline src/test/java/seedu/address/logic/commands/EditCommandTest.java:53]: Managed to work around this testcase as above for now, open to any improvement suggestions!

**Comment by wujiaweijoanna** [comment]
LGTM!

**Pull request merged**